services/dumper.js

```diff
@@ -132,7 +132,7 @@ function Dumper(config) {
       const expectedConventionalForeignKeyName = underscored
         ? _.snakeCase(reference.foreignKey) : reference.foreignKey;
       const foreignKeyColumnUnconventional =
-        reference.foreignKey !== expectedConventionalForeignKeyName;
+        reference.foreignKeyName !== expectedConventionalForeignKeyName;
```

Comment:
We mark a field as unconventional if it's not the same as foreignKeyName (To avoid the previous bug you had)

Reply:
We should have tests created to protect this code against regressions according to our new rules.